manifests/init.pp

```diff
@@ -231,6 +236,39 @@
     $config_file_ensure = present
   }
 
+  $manage_networking_ipv6 = $networking_ipv6 ? {
+    'yes'   => 'yes',
+    'no'    => 'no',
+    default => undef,
+  }
+  $manage_ipv6forwarding = $ipv6forwarding ? {
+    'yes'   => 'yes',
+    'no'    => 'no',
+    default => undef,
+  }
+  $manage_ipv6_autoconf = $ipv6_autoconf ? {
+    'yes'   => 'yes',
+    'no'    => 'no',
+    default => undef,
+  }
+  $manage_ipv6_autotunnel = $ipv6_autotunnel ? {
+    'yes'   => 'yes',
+    'no'    => 'no',
+    default => undef,
+  }
+  if $ipv6_defaultgw {
+    if $ipv6_defaultgw =~ Stdlib::IP::Address::V6 {
```

Comment:
This module is still compatible with old versions of Puppet where data types are not used.
This change can get through in the next major release.

Reply:
replaced with 

$manage_ipv6_defaultgw = $ipv6_defaultgw

without checks. IPv6 regexp is too long to do verification. 

Also i fond out recently. 
Placing gateway into /etc/sysconfig/network for IPv4 work as expected on netwtok-scripts. Interfaces configured and then GW set (once).
Then i tried same with NM it result in 0.0.0.0/0 route added to multiple active interfaces. I had to move GATEWAY= to interface definition file. 

So i think use of GATEWAY= option in the /etc/sysconfig/network will be deprecated or not in use.  

Let me know if this block is compatible as well, i've added it because it was listed in available options although i am not using it:

  if $ipv6_radvd_pidfile =~ Stdlib::Absolutepath {
    $manage_ipv6_radvd_pidfile = $ipv6_radvd_pidfile
  } else {
    $manage_ipv6_radvd_pidfile = undef
  }